File: android/mobilepasssdk/src/main/java/com/armongate/mobilepasssdk/model/request/RequestAnalyticsData.java
package com.armongate.mobilepasssdk.model.request;

import com.armongate.mobilepasssdk.enums.AnalyticsResult;
import com.armongate.mobilepasssdk.enums.AnalyticsOS;
import com.armongate.mobilepasssdk.enums.PassMethod;
import com.armongate.mobilepasssdk.model.AnalyticsStep;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class RequestAnalyticsData {
    @SerializedName("accessTime")
    public String accessTime;
    
    @SerializedName("duration")
    public long duration;
    
    @SerializedName("result")
    public String result;
    
    @SerializedName("method")
    public String method;
    
    @SerializedName("clubId")
    public String clubId;

    @SerializedName("qrCodeId")
    public String qrCodeId;
    
    @SerializedName("direction")
    public Integer direction;
    
    @SerializedName("os")
    public String os;
    
    @SerializedName("steps")
    public List<AnalyticsStep> steps;

    public RequestAnalyticsData(String accessTime, long duration, AnalyticsResult result,
                              PassMethod method, String clubId, String qrCodeId, Integer direction,
                              List<AnalyticsStep> steps) {
        this.accessTime = accessTime;
        this.duration = duration;
        this.result = result.getValue();
        this.method = method != null ? method.getValue() : null;
        this.clubId = clubId;
        this.qrCodeId = qrCodeId;
        this.direction = direction;
        this.os = AnalyticsOS.ANDROID.getValue();
        this.steps = steps;
    }
}

File: ios/MobilePassSDK/Model/Request/RequestAnalyticsData.swift
//
//  RequestAnalyticsData.swift
//  MobilePassSDK
//
//  Created by Erinc Cakir on 4.03.2025.
//


struct RequestAnalyticsData: Codable {
    var accessTime: String
    var duration: Int64
    var result: AnalyticsResult
    var method: PassMethod?
    var clubId: String?
    var qrCodeId: String?
    var direction: Int?
    var os: AnalyticsOS
    var steps: [AnalyticsStep]
    
    var dictionary: [String: Any] {
        var dict: [String: Any] = [
            "accessTime": accessTime,
            "duration": duration,
            "result": result.rawValue,
            "os": os.rawValue,
            "steps": steps.map { [
                "c": $0.c,
                "t": $0.t,
                "m": $0.m
            ].compactMapValues { $0 }}
        ]
        
        if let method = method {
            dict["method"] = method.rawValue
        }
        if let clubId = clubId {
            dict["clubId"] = clubId
        }
        if let qrCodeId = qrCodeId {
            dict["qrCodeId"] = qrCodeId
        }
        if let direction = direction {
            dict["direction"] = direction
        }
        
        return dict
    }
}
